SpatialGDK/Source/SpatialGDK/Public/Interop/Connection/SpatialEventMessages.h

```diff
@@ -0,0 +1,418 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#pragma once
+
+#include "CoreMinimal.h"
+#include "SpatialCommonTypes.h"
+
+#include <WorkerSDK/improbable/c_worker.h>
+
+#include "SpatialEventMessages.generated.h"
+
+static_assert(sizeof(Worker_EntityId) == sizeof(int64), "EntityId assumed 64-bit here");
+static_assert(sizeof(Worker_ComponentId) == sizeof(uint32), "ComponentId assumed 32-bit here");
+static_assert(sizeof(VirtualWorkerId) == sizeof(uint32), "VirtualWorkerId assumed 32-bit here");
+static_assert(sizeof(TraceKey) == sizeof(int32), "TraceKey assumed 32-bit here");
+static_assert(sizeof(Worker_RequestId) == sizeof(int64), "RequestId assumed 64-bit here");
+
+#define GDK_EVENT_NAMESPACE "unreal_gdk."
+
+// Tagged with cause - could
+USTRUCT()
+struct FEventMessage
+{
+	GENERATED_BODY()
+
+	FEventMessage() = default;
+	FEventMessage(const char* InType)
+		: Type(InType)
+	{
+	}
+
+	const char* GetType() const { return Type; }
+
+private:
+	const char* Type = "Null";
+};
+
+USTRUCT()
+struct FEventGenericMessage : public FEventMessage
+{
+	GENERATED_BODY()
+
+	FEventGenericMessage()
+		: FEventMessage("generic_message")
+	{
+	}
+	FEventGenericMessage(const FString& Message)
+		: FEventMessage(GDK_EVENT_NAMESPACE "generic_message")
+		, Message(Message)
+	{
+	}
+
+	UPROPERTY() FString Message;
+};
+
+// Tagged with cause
+USTRUCT()
+struct FEventCreateEntity : public FEventMessage
+{
+	GENERATED_BODY()
+
+	FEventCreateEntity()
+		: FEventMessage(GDK_EVENT_NAMESPACE "create_entity")
+	{
+	}
+	FEventCreateEntity(int64 EntityId, const AActor* Actor)
+		: FEventMessage(GDK_EVENT_NAMESPACE "create_entity")
+		, EntityId(EntityId)
+		, Actor(Actor)
+	{
+	}
+
+	UPROPERTY() int64 EntityId = -1;
+	UPROPERTY() const AActor* Actor = nullptr;
+};
+
+// Tagged with cause
+USTRUCT()
+struct FEventRemoveEntity : public FEventMessage
+{
+	GENERATED_BODY()
+
+	FEventRemoveEntity()
+		: FEventMessage(GDK_EVENT_NAMESPACE "remove_entity")
+	{
+	}
+	FEventRemoveEntity(int64 EntityId, const AActor* Actor)
+		: FEventMessage(GDK_EVENT_NAMESPACE "remove_entity")
+		, EntityId(EntityId)
+		, Actor(Actor)
+	{
+	}
+
+	UPROPERTY() int64 EntityId{ -1 };
```

Comment:
Minor nit that comes out throughout - for primitive and psuedo primitive types I think it's more common to use an = than an initialiser. That said this isn't wrong so if you disagree then feel free to disregard this.